app/models/agents/slack_agent.rb

```diff
@@ -67,7 +71,11 @@ def slack_notifier
     def receive(incoming_events)
       incoming_events.each do |event|
         opts = interpolated(event)
-        slack_notifier.ping opts[:message], channel: opts[:channel], username: opts[:username]
+        if /^:/.match(opts[:icon])
+            slack_notifier.ping opts[:message], channel: opts[:channel], username: opts[:username], icon_emoji: opts[:icon]
+        else
```

Comment:
Okay, that works for me.  If you wouldn't mind adding a spec or two to slask_agent_spec.rb and fixing the indent, this seems great to me!